### PR TITLE
Suggest other regions

### PR DESCRIPTION
**Pull request by viraptor**
When trying to apply a stack in a region where it's not defined we could do
better than just print the error.
List regions where the stack by that name is defined.

### CODE REVIEW

File: lib/stack_master/cli.rb [hunk 1]
```diff
@@ -270,6 +271,13 @@ def execute_stacks_command(command, args, options)
       @kernel.exit false unless success
     end
 
+    def show_other_region_candidates(stack_name)
+      candidates = config.filter(stack_name=stack_name)
+      return if candidates.empty?
+
+      StackMaster.stdout.puts "Stack name #{stack_name} exists in regions: #{candidates.map(:&region).join(', ')}"
```
Comment (simpsora) [inline lib/stack_master/cli.rb]: I wonder if this should be more clearly worded as an error case, with a helpful suggestion.  e.g.:

```suggestion
      StackMaster.stdout.puts "Stack name #{stack_name} not found in region #{region}, but was found in regions: #{candidates.map(:&region).join(', ')}"
```

Comment (viraptor) [reply lib/stack_master/cli.rb]: This is already a part of the previous line: https://github.com/envato/stack_master/pull/343/files#diff-f83645dd9bf35512a532426db4fed419R256

Comment (simpsora) [reply lib/stack_master/cli.rb]: Oh -- so it does.

**Comment by simpsora** [comment]
Have merged https://github.com/envato/stack_master/pull/342, so feel free to merge that in here, update the docs, and we can release both at once.

File: features/apply.feature [hunk 2]
```diff
@@ -150,6 +150,11 @@ Feature: Apply command
     Then the output should contain "Could not find stack definition bar in region foo"
     And the exit status should be 1
 
+  Scenario: Run apply with stack in wrong region
+    When I run `stack_master apply foo myapp-web`
+    Then the output should contain "Stack name myapp-web exists in regions: us-east-1"
+    And the exit status should be 1
+
```
Comment (simpsora) [inline features/apply.feature:157]: This is interesting, I opted for unit tests rather than features.  I wonder which is better 🤔

Comment (stevehodgkiss) [reply features/apply.feature:157]: This level of testing can also be done via rspec+aruba. I opted for that approach in the drift feature and I'm wondering if we should re-write all the cucumber features to use rspec instead. https://github.com/envato/stack_master/pull/339/files#diff-0845fa726c236bab4b7365489a3cd108R1

Comment (viraptor) [reply features/apply.feature:157]: In this case there was an existing cucumber test for the output of this command, so it was the easiest solution.

Comment (stevehodgkiss) [reply features/apply.feature:157]: Yep, 👍. Sorry, I wasn't suggesting that this change needed to move to rspec.

**Pull request merged**